lib/nats.js

```diff
@@ -843,6 +865,9 @@ Client.prototype.publish = function(subject, msg, opt_reply, opt_callback) {
  */
 
 Client.prototype.subscribe = function(subject, opts, callback) {
+  if (this.closed) {
+    this.emit('error', SUB_CONN_CLOSED_ERR);
```

Comment:
Always interested in good ideas and I think you and @ronag have better stylistic understandings of Node.js.

For the emit though, didn't we say that we should not direct to the connection handler for localized events? The connection closed on is special since if the connection is closed, all event listeners have been removed and the error will be thrown anyway, no way to avoid that as the code is written today.